Content/LeagueSandbox-Scripts/Buffs/Global/Stun.cs

```diff
@@ -11,25 +11,33 @@ internal class Stun : IBuffGameScript
         public BuffType BuffType => BuffType.STUN;
         public BuffAddType BuffAddType => BuffAddType.REPLACE_EXISTING;
         public int MaxStacks => 1;
-        public bool IsHidden => true;
+        public bool IsHidden => false;
 
         public IStatsModifier StatsModifier { get; private set; }
 
+        IAttackableUnit owner;
         IParticle stun;
 
         public void OnActivate(IAttackableUnit unit, IBuff buff, ISpell ownerSpell)
         {
-            stun = AddParticleTarget(ownerSpell.CastInfo.Owner, null, "Global_Stun.troy", unit, buff.Duration, bone: "head");
+            owner = unit;
+            SetStatus(unit, StatusFlags.Stunned, true);
+            stun = AddParticleTarget(ownerSpell.CastInfo.Owner, unit, "LOC_Stun.troy", unit, buff.Duration, bone: "head");
         }
 
         public void OnDeactivate(IAttackableUnit unit, IBuff buff, ISpell ownerSpell)
         {
+            owner = null;
+            SetStatus(unit, StatusFlags.Stunned, false);
             RemoveParticle(stun);
         }
 
         public void OnUpdate(float diff)
         {
-
+            if (owner != null)
+            {
+                SetStatus(owner, StatusFlags.Stunned, true);
```

Comment:
Why is it needed? Shouldn't the unit itself handles if it is stunned by some sort of smart tracking of buffs?

Reply:
The main reason for this was just replicating the functionality of league's scripts.
I don't believe it will cause race conditions as the update loop is a single loop, not multi threaded.

I will change it for now as your suggestion is at least more intuitive.